Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -979,10 +1005,18 @@ void FastStringifier::append(JSValue value)
             recordBufferFull();
             return;
         }
-        char* cursor = reinterpret_cast<char*>(m_buffer) + m_length;
-        auto result = std::to_chars(cursor, cursor + maxInt32StringLength, number);
-        ASSERT(result.ec != std::errc::value_too_large);
-        m_length += result.ptr - cursor;
+        if constexpr (sizeof(CharType) == 1) {
+            char* cursor = bitwise_cast<char*>(m_buffer) + m_length;
+            auto result = std::to_chars(cursor, cursor + maxInt32StringLength, number);
+            ASSERT(result.ec != std::errc::value_too_large);
+            m_length += result.ptr - cursor;
+        } else {
+            std::array<char, maxInt32StringLength> temporary;
+            auto result = std::to_chars(temporary.data(), temporary.data() + maxInt32StringLength, number);
+            ASSERT(result.ec != std::errc::value_too_large);
+            WTF::copyElements(bitwise_cast<uint16_t*>(&m_buffer[m_length]), bitwise_cast<const uint8_t*>(temporary.data()), result.ptr - temporary.data());
+            m_length += result.ptr - temporary.data();
```

Comment:
nit: precompute the `lengthToCopy` just once so that the code shows that both of these are using the same value?

Reply:
Done.